confd/confd.toml

```diff
@@ -0,0 +1,3 @@
+confdir = "/etc/confd"
+backend = "file"
+file = [ "/etc/confd/opennms.yaml" ]
```

Comment:
@agalue Just to be sure here. We require here to use configuration file and do not allow ENV vars. The original suggestion was to default to ENV and allow the user to switch the backend if he wants to something different. This would not require additional logic overwrite confd configurations. 

```
ENV CONFD_BACKEND="env"
```

To use a file we just could set the CONFD_BACKEND at runtime to `file -file /etc/confd/opennms.yaml`. The change [here](https://github.com/opennms-forge/docker-horizon-core-web/pull/58/commits/c6f5804d879fd29a9270be8294ef79c0ab0093d7) removed that flexibility to [use other backends](https://github.com/kelseyhightower/confd/blob/master/docs/quick-start-guide.md) as well.

Reply:
There is no need for `CONFD_BACKEND`. If you want to change the backend, just create your own `confd.toml` and mount it to `/etc/confd`. That way you can use something else like etcd, consul, zookeeper, etc.

The reason why `env` won't work any more is because you can't use camel-case, or '-' or '_' on the key paths, which is not a problem with all the rest of the backends (including the default: `file`). Nothing prevents you to use it, but it just won't work.

Reply:
In terms of just override the YAML file with your own configuration, you're write, just mount the desired YAML to `/etc/confd/opennms.yaml`.

Reply:
ah you mean keys like these here: `/map-geocoding`. Why not using than just `/map/geocoding`? I'm nagging a bit cause the docker smoke tests use for these basic things with ENV_VARS. But as I remember @mattixtech did some of the new smoke tests and he is fine with it, I'm fine with it too :)

Reply:
Read the whole discussion from the beginning ;)

The property is called `map-geocoding` which is why using that exact same format makes sense, even if that will never work with the `env` backend (but works for all the rest of them).

Now, there are cases when you can find `mapGeoCoding` or `map_geocoding` (I'm just using the same *term* as an example, not saying those are real cases). Both options work with *all* backends except for `env`.

I've decided to keep the key paths as close as possible as the original properties, even if there is no structure, name convention or consistency among those properties.

If we want consistency, we need to refactor OpenNMS. Then, continue working on this PR. Or, leave OpenNMS as it is, use the current state of this PR. Then, when we *normalize* the property names, we can fix the key paths, and add an *upgrade* section for future versions of the Docker image.

There are lots of possibilities, all make sense, but we should make a decision in order to merge this PR. This reminds me, there are a few changes on `release/24.0.0-rc` (for OpenNMS, Minion and Sentinel), that worth migrating to this one.

Reply:
> ah you mean keys like these here: `/map-geocoding`. Why not using than just `/map/geocoding`? I'm nagging a bit cause the docker smoke tests use for these basic things ENV_VARS. But as I remember @mattixtech did some of the new smoke tests and if he is fine with it, I'm fine with it too :)

From my point of view for smoke-tests, its not that big of a deal to move away from env. We simply maintain a yaml file with the properties we want for that test and mount it to the container for the test run. I would prefer this method for all but the most trivial configuration since adding a bunch of env variables to the container in that code is a bit nasty in my opinion.

Dropping support for env backend lets us have much more freedom over how we want to structure/name our keys and does away with the issue where we have to translate form a key path to an env variable (by uppercasing, replacing '/' with '_' etc.).